lib/src/http/common/request.dart

```diff
@@ -767,11 +767,11 @@ abstract class CommonRequest extends Object
       checkForTimeout();
       final responseCompleter = Completer<BaseResponse>();
 
-      // Enforce a timeout threshold if set.
-      Timer timeout;
-      if (timeoutThreshold != null) {
-        timeout = Timer(autoRetry.timeoutThreshold, _timeoutRequest);
+      // Enforce a timeout threshold.
+      if (timeoutThreshold == null) {
+        timeoutThreshold = defaultTimeoutThreshold;
       }
+      final timeout = Timer(autoRetry.timeoutThreshold, _timeoutRequest);
```

Comment:
Someone could set `defaultTimeoutThreshold` to null, so we should still guard against that here.